packages/mui-material/src/styles/createTheme.spec.ts

```diff
@@ -81,3 +81,41 @@ const theme = createTheme();
     },
   });
 }
+
+{
+  createTheme(theme, {
+    components: {
+      MuiButton: {
+        variants: [
+          {
+            props: {}, // match any props combination
+            style: ({ theme: t }) => {
+              return {
+                fontFamily: t.typography.fontFamily,
+              };
+            },
+          },
+        ],
+      },
+    },
+  } as ThemeOptions);
```

Comment:
the second argument of `createTheme` is an object, so if we want to improve we need to cast.